ssl/record/rec_layer_s3.c

```diff
@@ -1135,7 +1135,7 @@ int ssl_set_new_record_layer(SSL_CONNECTION *s, int version,
     int use_etm, stream_mac = 0, tlstree = 0;
     unsigned int maxfrag = SSL3_RT_MAX_PLAIN_LENGTH;
     int use_early_data = 0;
-    uint32_t max_early_data;
+    unsigned int max_early_data;
```

Comment:
Why this change? This is not correct, `ossl_get_max_early_data()` returns `uint32_t`.

Reply:
But OSSL_PARAM_construct_uint() expects pointer to unsigned int. IMO the change is right. At worst the unsigned int will be 64 bit (we do not support systems with 16 bit integers) and this will be implicitly cast from uint32_t.

Reply:
@hlandau please reconsider approval

Reply:
Hold on, if we don't support systems with 16 bit integers then why does this need to be changed in the first place?

Reply:
64-bit integers?

Reply:
> Hold on, if we don't support systems with 16 bit integers then why does this need to be changed in the first place?

Because `uint32` is defined as `unsigned long int` on djgpp, and the compiler considers that a different type from `unsigned int`, even though both are 32-bit.  So the pointer conversion for `OSSL_PARAM_construct_uint` causes a warning.

Reply:
> I see now there is also `OSSL_PARAM_construct_uint32` - maybe that should be used instead?

That would be a solution.